src/lib.rs

```diff
@@ -454,7 +454,7 @@ pub struct Type {
 }
 
 /// Enum with additional information, depending on the kind of type.
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Clone)]
```

Comment:
Is there anywhere the clone is necessary vs. a reference?

Reply:
This is used to provide better error messages for Error::BadTypeCast. I can unfortunately not use a reference because the struct may even be temporary. Fortunately it is only used in the error path, in which performance is less of an issue.
I did worry a bit about using it in BadTypeCast though because the TypeInner struct is quite large. Possibly they should be boxed to avoid copying costs?

![image](https://user-images.githubusercontent.com/1144597/120019565-e7f76480-bfe8-11eb-8899-6895082ae219.png)

Reply:
Let's try to find another way. For example, these could be strings, output from `type_to_name`.
We don't want to have `TypeInner` cloneable in general, because it invites for inefficient code paths to appear.

Reply:
Ok, I changed those to strings.